### PR TITLE
Add numpy as dependency

### PR DESCRIPTION
**Pull request by fran6co**
Fixes https://github.com/triton-inference-server/dali_backend/issues/142

### CODE REVIEW

File: cmake/dalienv.yml.in [hunk 1]
```diff
@@ -2,6 +2,7 @@ name: dalienv
 dependencies:
   - python=3.8
   - pip
+  - numpy
```
Comment (JanuszL) [inline cmake/dalienv.yml.in:5]: I'm sorry to say that it is not a good change.
First of all, DALI doesn't depend on the numpy, you can just build a flow that uses numpy as an auxiliary tool but it is not a must.
This way we would need to add a dependency to Torch and CuPy as DALI can use tensors from these frameworks.

Comment (fran6co) [reply cmake/dalienv.yml.in:5]: I added a specific use case in the linked issue, could you recommend how to solve it without numpy?

Comment (JanuszL) [reply cmake/dalienv.yml.in:5]: In this case I believe you need numpy to create the serialized pipeline you would later use in during the inference through the DALI backend. During the inference process itself, the numpy package is not needed, so you can use any environment with numpy, not necessarily inside the TRITON dev image.

**Comment by szalpal** [comment]
Since the original issue was resolved without the need of using `numpy`, I'm closing this PR. However, adding extra Python deps is a valid point, please refer to the original issue for my further explanation: https://github.com/triton-inference-server/dali_backend/issues/142#issuecomment-1193715288

**Pull request closed without merging**